backend/database/bespoke.ts

```diff
@@ -180,4 +182,65 @@ export class BespokeQueries {
 
     return value[0][Object.keys(value[0])[0]];
   }
+
+  static async getReturnBalanceItemsQty(
+    db: DatabaseCore,
+    schemaName: string,
+    docName: string
+  ): Promise<DocValueMap[] | undefined> {
+    const docItems = (await db.knex!(`${schemaName}Item`)
+      .select('item')
+      .where('parent', docName)
+      .groupBy('item')
+      .sum({ quantity: 'quantity' })) as DocValueMap[];
+
+    const returnDocNames = (
+      await db.knex!(schemaName)
+        .select('name')
+        .where('returnAgainst', docName)
+        .andWhere('submitted', true)
+        .andWhere('cancelled', false)
+    ).map((i: { name: string }) => i.name);
+
+    if (!returnDocNames.length) {
+      return;
+    }
+
+    const returnedItems = (await db.knex!(`${schemaName}Item`)
+      .select('item')
+      .sum({ quantity: 'quantity' })
+      .whereIn('parent', returnDocNames)
+      .groupBy('item')) as DocValueMap[];
+
+    if (!returnedItems.length) {
+      return;
+    }
+
+    const returnBalanceItemQty = [];
```

Comment:
Why isn't this typed?

If you're calling something returnBalanceItem**Qty** why are you inserting the entire item along with the quantity.

Along with very weak typing, the function returns `DocValueMap[]` where as the important piece of information is the `quantity` this should be captured in the type being returned.

Reply:
Okay on checking again, you're passing just `{item: string; quantity: number}` you could've typed it as such, or more preferably you could've just returned a map: `Record<Item, Quantity>`